app/src/main/java/com/heapixLearn/discovery/logic/contact/LogicContactManager.java

```diff
@@ -0,0 +1,103 @@
+package com.heapixLearn.discovery.logic.contact;
+
+import android.os.Handler;
+import android.os.Looper;
+
+import com.heapixLearn.discovery.logic.contact.contacts_domain_interfaces.DBStoreContactManager;
+import com.heapixLearn.discovery.logic.contact.contacts_domain_interfaces.ServerStoreContactManager;
+import com.heapixLearn.discovery.logic.contact.contacts_domain_interfaces.UIContactManager;
+
+import java.util.ArrayList;
+
+public class LogicContactManager {
```

Comment:
@dianaprimshits in this class you use a lot of threads which is resource consuming approach, it can be optimized with handler

Reply:
@YodaMike I use handler. Isn't it?

Reply:
No. You use handler to run runnables that you get in the methods, but all your threads can run simultaneously